frontend/RaffleBay/RaffleBay/View/ContentView.swift

```diff
@@ -8,33 +8,112 @@
 
 import SwiftUI
 
+//Creation of our own Navigation View
+struct NavigationItem{
+   var view: AnyView
+}
+
+final class NavigationStack: ObservableObject {
+    @Published var viewStack: [NavigationItem] = []
+    @Published var currentView: NavigationItem
+    
+    init(_ currentView: NavigationItem ){
+       self.currentView = currentView
+    }
+    
+    func unwind(){
+       if viewStack.count == 0{
+          return }
+       let last = viewStack.count - 1
+       currentView = viewStack[last]
+       viewStack.remove(at: last)
+    }
+    
+    func advance(_ view:NavigationItem){
+       viewStack.append( currentView)
+       currentView = view
+    }
+    
+    func home( ){
+       currentView = NavigationItem( view: AnyView(SaleItemTableView()))
+       viewStack.removeAll()
+    }
+    
+    func splashscreen( ){
+       currentView = NavigationItem( view: AnyView(SplashscreenView()))
+       viewStack.removeAll()
+    }
+    
+    func login( ){
+       currentView = NavigationItem( view: AnyView(LoginView()))
+       viewStack.removeAll()
+    }
+    
+    func signup( ){
+       currentView = NavigationItem( view: AnyView(SignupView()))
+       viewStack.removeAll()
+    }
+    
+    func success( ){
+       currentView = NavigationItem( view: AnyView(SuccessfulView()))
+       viewStack.removeAll()
+    }
+    
+    
+}
+
+struct NavigationHost: View{
+   @EnvironmentObject var navigation: NavigationStack
+   
+   var body: some View {
+      self.navigation.currentView.view
+   }
+}
+
+//Check if we have some auth token
+func whatView(token: String) -> AnyView{
+//    if (token == ""){
```

Comment:
is this supposed to be uncommented? Looks like when i restart the app, it goes back to the login page instead of remembering my details

Reply:
I need to look back into this. Yes it should be uncommented but I also believe there was so glitch here but should be a quick fix.